### PR TITLE
 Module 6 assignments

### PR DESCRIPTION
**Pull request by JuleeGunjal**

### CODE REVIEW

File: module_6/array_operations.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                "email": "tpetyt9@upenn.edu",
+                "date_of_birth": "2018/09/01"
+              }]
+
+// Write a function filterByName that accepts a string as a parameter 
+// and returns an array with only those objects where the first_name field includes that string.
+
+let filterByName = (nameParam) => {
+  let filterArray = array.filter(item => item.first_name.includes(nameParam));
+  return filterArray;
```
Comment (tejashreeSurveJosh) [inline module_6/array_operations.js]: You can directly return the result of the array.filter

File: module_6/array_operations.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+//     first_name: 'Venita',
+//     email: 'vheap4@clickbank.net',
+//     date_of_birth: '2020/10/04'
+//   }
+// ]
+
+// Use Array.map to return an array of all the email fields.
+
+let getEmails = () => {
+  let emailArray = array.map(item =>item.id + "_email: " + item.email);
```
Comment (tejashreeSurveJosh) [inline module_6/array_operations.js]: Here use can destructure all the value that you want to use 
const {is,email} = item;

File: module_6/array_operations.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+// Use Array.sort to return the array sorted in descending order by date_of_birth.
+
+let sortByBirthDateDesc = () => {
+  let sortedArray = array.sort((a, b) => sortFunction(a, b));
+  return sortedArray;
+}
+
+let sortFunction = (a, b) => {
+  if(a.date_of_birth > b.date_of_birth) return 1;
```
Comment (tejashreeSurveJosh) [inline module_6/array_operations.js]: here also use can destructure date of birth with other name

File: module_6/array_operations.js (hunk 1 above)
Comment (JuleeGunjal) [reply module_6/array_operations.js]: done

File: module_6/array_operations.js (hunk 2 above)
Comment (JuleeGunjal) [reply module_6/array_operations.js]: Done

File: module_6/array_operations.js (hunk 3 above)
Comment (JuleeGunjal) [reply module_6/array_operations.js]: Done

**Pull request merged**